### ISSUE TITLE
Pausing download makes download unrecoverable 

### ISSUE THREAD

**Issue by ballo** [TeamNewPipe/NewPipe#11076]
### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

27

### Steps to reproduce the bug

1. Initiate download 
2. Pause download
3. Do other stuff like use a web browser
4. Try to resume download

### Expected behavior

NewPipe will delete the paused download if it gets swapped out of ram. The file remains, but NewPipe can only overwite it rather than resume. This doesn't appear to be a problem if the download is interrupted. In this case. It will change the status of the download in the notifications and will always allow the download to be resumed. The issue is if you pause the download. 

Current work-arounds:
1. Disconnect data to cause the download interrupted error which works unlike pausing the download 
2. Use something else like yt-dlp (my current solution)

**Comment by ballo** [comment]
> To be clear, the app handles ejection from RAM properly for an interrupted download, but not a paused one?
That's the behavior I witnessed, yes

**Comment by NewNewPipe** [comment]
Nihil novi...
Old problem
https://github.com/TeamNewPipe/NewPipe/issues/7059

**Comment by liloman** [comment]
I found the bug on DownloadManager.loadPendingMissions: 

````java
if (mis == null || mis.isFinished() || mis.hasInvalidStorage()) {
                //noinspection ResultOfMethodCallIgnored
                sub.delete();
                continue;
            }
   mis.threads = new Thread[0];

  boolean exists;
            try {
                mis.storage = StoredFileHelper.deserialize(mis.storage, ctx);
                exists = !mis.storage.isInvalid() && mis.storage.existsAsFile();
            } catch (Exception ex) {
                Log.e(TAG, "Failed to load the file source of " + mis.storage.toString(), ex);
                mis.storage.invalidate();
                exists = false;
            }

 ````
 
In this commit mis.hasInvalidStorage() is called before mis.storage being deserialized (resulting in always failing in checking if "mis.storage.existsAsFile" ) and hence executing sub.delete(); (deleting the pending downloads) 

https://github.com/TeamNewPipe/NewPipe/pull/6778/commits/1822d216763b92b563179e2a68c0c2bbb2020870

The solution is just remove that redundant check that was done after anyway:

````java
if (mis == null || mis.isFinished() ) {
                //noinspection ResultOfMethodCallIgnored
                sub.delete();
                continue;
            }
...
 ````

It's quite a nasty bug because many people use mainly the download function and till version 0.21.8 was working properly (2021).

**Comment by Stypox** [comment]
@liloman sorry for the late response, and thank you for investigating! Could you open a pull request? Otherwise let me know, I can open one myself. :-)

By the way, we are receiving many reports of this bug in the email, too.

**Comment by Alvenereo** [comment]
I get this bug every day. How annoying. I hope they fix it in the next update. Cheers.

**Comment by liloman** [comment]
> [@liloman](https://github.com/liloman) sorry for the late response, and thank you for investigating! Could you open a pull request? Otherwise let me know, I can open one myself. :-)
> 
> By the way, we are receiving many reports of this bug in the email, too.

Go on, no problem.

I hardly use github ...

**Comment by ZeddZoul** [comment]
This bug is nasty. Messed me up today. My download completed tho. But the parsing after download was affected. I'm hoping i can resume the download

**Comment by robzorua** [comment]
Correction for the current work-arounds number 1 it doesn't work anymore for v0.28.0. Downloaded with interrupted error status will be cleared automatically in download list if force stop. The remnant of the incomplete download will reside on the phone's memory.

**Comment by ShareASmile** [comment]
Please test the APK of #12550 and report whether it fixes the problem.

**Comment by ballo** [comment]
yeah unfortunately there is no way to resume when this happens. I've gone back to yt-dlp despite the tedious steps. at least partials are recoverable

**Comment by liloman** [comment]
I just tested latest release 0.28.0:

1-Close app while downloading  (works cause download continues on background) 

2-Pause the download and then close de app (doesn't work, doesn't delete the file but it is deleted from the download queue)

I have tried my patch, just 1 line, and It worked as expected.
